packages/astro/src/core/build/generate.ts

```diff
@@ -156,15 +171,24 @@ async function generatePage(
 	// Get paths for the route, calling getStaticPaths if needed.
 	const paths = await getPathsForRoute(pageData, pageModule, opts, builtPaths);
 
-	for (let i = 0; i < paths.length; i++) {
-		const path = paths[i];
-		await generatePath(path, opts, generationOptions);
-		const timeEnd = performance.now();
-		const timeChange = getTimeStat(timeStart, timeEnd);
-		const timeIncrease = `(+${timeChange})`;
-		const filePath = getOutputFilename(opts.astroConfig, path, pageData.route.type);
-		const lineIcon = i === paths.length - 1 ? '└─' : '├─';
-		info(opts.logging, null, `  ${cyan(lineIcon)} ${dim(filePath)} ${dim(timeIncrease)}`);
+	const concurrency = opts.buildConfig.concurrency || MAX_CONCURRENT_RENDERS;
+	if (concurrency > 1) {
```

Comment:
It would be nice if the codepaths didn't diverge this month. Is there a reason why we can't always go through the generatePathsConcurrently path but concurrency defaults to 1?

Reply:
Mostly it was because the pretty line icon does not work in concurrent mode (no notion of first / last when printing the line), but if we don't care about those and always can use `-` then those could be merged.

Reply:
I removed the alternative code path, and in order to close up the "icon tree" I added a summary line like this:
```
▶ src/pages/buy/[slug].astro
⚠️  Building with 8 concurrent renders
  ├─ /buy/foo/index.html (+28ms)
  ├─ /buy/bar/index.html (+45ms)
  ├─ /buy/baz/index.html (+45ms)
  └─ Completed src/pages/buy/[slug].astro in 45ms
```